### PR TITLE
Add pnor layout doc

### PR DESCRIPTION
**Pull request by IgorBagnucki**

### CODE REVIEW

**Comment by macpijan** [comment]
@IgorBagnucki Still unverifeid commits. You are signing with `402E2B7A9B71D1BF`, but you have not added this key to GitHub.

File: devnotes/pnor_layout.md [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+Partition Table
+Version: 1
+Size: 2 * 0x1000 (block size)
+Block Count: 0x4000
+Entry Size: 0x80
+Number of entries: 0x21
+
+| Name                        | ID | Offset:End         | Size    | Type | Flags | Actual size  | coreboot usage              |
+| --------------------------- | -- | ------------------ | -------:| ---- | ----- | ------------:| --------------------------- |
+| [part](#part)               | 01 | 00000000:00001FFF  | 2000    | p    | p     | same as size | partition table             |
```
Comment (macpijan) [inline devnotes/pnor_layout.md:60]: I assume the size is in hex and then bytes? Let's provide such information at least. Maybe human readable unit would be more useful?

File: devnotes/pnor_layout.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+| [GUARD](#GUARD)             | 03 | 0002C000:00030FFF  | 5000    | d    | -     | same as size | _unknown_                   |
+| NVRAM                       | 04 | 00031000:000C0FFF  | 90000   | d    | -     | same as size | _unknown_                   |
+| [SECBOOT](#SECBOOT)         | 05 | 000C1000:000E4FFF  | 24000   | d    | -     | same as size | _unknown_                   |
+| DJVPD                       | 06 | 000E5000:0012CFFF  | 48000   | d    | -     | same as size | _unknown_                   |
+| MVPD                        | 07 | 0012D000:001BCFFF  | 90000   | d    | -     | same as size | _unknown_                   |
+| CVPD                        | 08 | 001BD000:00204FFF  | 48000   | d    | -     | same as size | _unknown_                   |
+| [HBB](#HBB)                 | 09 | 00205000:00304FFF  | 100000  | d    | -     | same as size | bootblock                   |
+| HBD                         | 0A | 00305000:00424FFF  | 120000  | d    | -     | same as size | _unknown_                   |
+| [HBI](#HBI)                 | 0B | 00425000:019C4FFF  | 15A0000 | d    | -     | same as size | romstage, ramstage, payload |
```
Comment (macpijan) [inline devnotes/pnor_layout.md:70]: Maybe be explicit that payload is skiboot: `payload (skiboot)`.

File: devnotes/pnor_layout.md [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+Probably just an information about versions of a binaries in the PNOR image.
+
+## HBBL
+This partition is used by hostboot during update operation.
+
+## IMA_CATALOG
+This partition holds a list of events supported by OCC microcode.
+
+## BOOTKERNFW
+Probably `petitboot`, `initramfs`.
```
Comment (macpijan) [inline devnotes/pnor_layout.md:156]: @IgorBagnucki According to the config you have dumped, the initramfs is bundled with kernel image: https://github.com/3mdeb/openpower-coreboot-docs/pull/41/files#diff-927e415b44f07619c187353e91eed8ebfa02a5489973f5c93184653387f8782cR134

So there is a single kernel image which contains the initramfs already. And the initramfs contains the petitboot binaries of course. So all of that is in the `BOOTKERNEL` partition

File: devnotes/pnor_layout.md [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+
+## SECBOOT
+This partition is used to hold secure variables across renppts in
+tamper-resistant manner. Because writes to PNOR can't be prevented, hash of
+this partition is stored in the `TPM NV`, and verified at the boot time.
+More informations can be found at the
+[wiki page](https://open-power.github.io/skiboot/doc/secvar/secboot_tpm.html).
+
+## HBB
+This partition is overwritten with `coreboot` image containing `bootblock`.
```
Comment (macpijan) [inline devnotes/pnor_layout.md:131]: @IgorBagnucki But what was the original usage? These sections are a mix of original purpose vs coreboot purpose. Maybe clearly define subsections in each of them and then describe what was the original purpose and what is the purpose in the coreboot case.

File: devnotes/pnor_layout.md [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+| BMC_INV                     | 14 | 03968000:03970FFF  | 9000    | d    | -     | same as size | _unknown_                   |
+| [HBBL](#HBBL)               | 15 | 03971000:03977FFF  | 7000    | d    | -     | same as size | _unknown_                   |
+| ATTR_TMP                    | 16 | 03978000:0397FFFF  | 8000    | d    | -     | same as size | _unknown_                   |
+| ATTR_PERM                   | 17 | 03980000:03987FFF  | 8000    | d    | -     | same as size | _unknown_                   |
+| [IMA_CATALOG](#IMA_CATALOG) | 18 | 03989000:039C8FFF  | 40000   | d    | -     | same as size | _unknown_                   |
+| RINGOVD                     | 19 | 039C9000:039E8FFF  | 20000   | d    | -     | same as size | _unknown_                   |
+| WOFDATA                     | 1A | 039E9000:03CE8FFF  | 300000  | d    | -     | same as size | _unknown_                   |
+| HB_VOLATILE                 | 1B | 03CE9000:03CEDFFF  | 5000    | d    | -     | same as size | _unknown_                   |
+| MEMD                        | 1C | 03cee000:03CFBFFF  | E000    | d    | -     | same as size | _unknown_                   |
```
Comment (macpijan) [inline devnotes/pnor_layout.md:87]: @IgorBagnucki There's a lot of `unknown` for coreboot. I think too much. We have written a substantial amount of coreboot code, so we should be able to tell if we use given partition or not. If it is currently unused (but we are not sure if we need it in the future) let's mark it as `currently unused`. Try to have as little unknows as possible.

For example , this one holds the VPD for RAM IIRC. So  we pobably already use it in coreboot? It also lacks description.

**Review by macpijan — CHANGES_REQUESTED** [review]

File: devnotes/pnor_layout.md (hunk 1 above)
Comment (IgorBagnucki) [reply devnotes/pnor_layout.md]: Information about hex and unit is now present.

File: devnotes/pnor_layout.md (hunk 2 above)
Comment (IgorBagnucki) [reply devnotes/pnor_layout.md]: Fixed

File: devnotes/pnor_layout.md (hunk 3 above)
Comment (IgorBagnucki) [reply devnotes/pnor_layout.md]: Fixed

File: devnotes/pnor_layout.md (hunk 4 above)
Comment (IgorBagnucki) [reply devnotes/pnor_layout.md]: I have moved this information to the table with partition layout and left here informations about original usage only

File: devnotes/pnor_layout.md [hunk 6]
```diff
@@ -0,0 +1,181 @@
+# About this document
+This document gathers known informations about partition layout in the original
+PNOR image from the Talos II platform.
+
+# Partition table
+
+PNOR contains partition table at offset 0. It is constructed out of header and multiple entries.
+Refer to the [ffs source code](https://github.com/open-power/ffs/blob/3ec70fbc458e32eef0d0b1de79688b4dc48cbd57/ffs/ffs.h#L108) for more detailed informations.
```
Comment (macpijan) [inline devnotes/pnor_layout.md:8]: @IgorBagnucki Pleas fix line wrap. `[ffs source code](https://github.com/open-power/ffs/blob/3ec70fbc458e32eef0d0b1de79688b4dc48cbd57/ffs/ffs.h#L108)` could be in a separate line

File: devnotes/pnor_layout.md [hunk 7]
```diff
@@ -0,0 +1,181 @@
+# About this document
+This document gathers known informations about partition layout in the original
```
Comment (macpijan) [inline devnotes/pnor_layout.md:2]: @IgorBagnucki There is no plural from `information`. Please use https://languagetool.org or different tool for improving documentation quality

File: devnotes/pnor_layout.md [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+Probably just an information about versions of a binaries in the PNOR image.
+
+## HBBL
+This partition is used by hostboot during update operation.
+
+## IMA_CATALOG
+This partition holds a list of events supported by OCC microcode.
+
+## MEMD
+Holds VPD for RAM IIRC.
```
Comment (macpijan) [inline devnotes/pnor_layout.md:175]: `IIRC` ? Let's make sure what's there

File: devnotes/pnor_layout.md (hunk 5 above)
Comment (IgorBagnucki) [reply devnotes/pnor_layout.md]: I have verified each partition and marked which ones are required and which are fine to be erased. It is however possible that some of them are still required in later stages or as some kind of storage mid-booting.

File: devnotes/pnor_layout.md (hunk 6 above)
Comment (IgorBagnucki) [reply devnotes/pnor_layout.md]: Fixed

File: devnotes/pnor_layout.md (hunk 7 above)
Comment (IgorBagnucki) [reply devnotes/pnor_layout.md]: Fixed

File: devnotes/pnor_layout.md [hunk 9]
```diff
@@ -0,0 +1,186 @@
...
+* l = FFS_TYPE_LOGICAL
+* d = FFS_TYPE_DATA
+* p = FFS_TYPE_PARTITION
+
+Flags:
+* b = FFS_FLAGS_U_BOOT_ENV
+* p = FFS_FLAGS_PROTECTED
+
+# Reproducibility
+Reproducibility of partitions can be found at the
```
Comment (macpijan) [inline devnotes/pnor_layout.md:111]: Rather, content (binary placed in the partition) can be reproducible, not the partition itself.

File: devnotes/pnor_layout.md [hunk 10]
```diff
@@ -0,0 +1,186 @@
...
+Self Boot Engine, stored in module EEPROM
+
+## HCODE
+FW for PPE
+
+## PAYLOAD
+This partition contains `skiboot`.
+
+## BOOTKERNEL
+This partition contains `skiroot`, `petitboot` and initramfs.
```
Comment (macpijan) [inline devnotes/pnor_layout.md:161]: This partition contains `skiroot`: Linux kernel bundled with `initrd` (Linux userspace and  `petitboot` binaries).

Comment (IgorBagnucki) [reply devnotes/pnor_layout.md]: Fixed

File: devnotes/pnor_layout.md (hunk 9 above)
Comment (IgorBagnucki) [reply devnotes/pnor_layout.md]: Fixed

File: devnotes/pnor_layout.md [hunk 11]
```diff
@@ -0,0 +1,189 @@
...
+| [HBBL](#HBBL)               | 0x15 | 0x03971000:0x03977FFF  | 0x7000       | d    | -     | same as size | None                                  |
+| ATTR_TMP                    | 0x16 | 0x03978000:0x0397FFFF  | 0x8000       | d    | -     | same as size | None                                  |
+| ATTR_PERM                   | 0x17 | 0x03980000:0x03987FFF  | 0x8000       | d    | -     | same as size | None                                  |
+| [IMA_CATALOG](#IMA_CATALOG) | 0x18 | 0x03989000:0x039C8FFF  | 0x40000      | d    | -     | same as size | None                                  |
+| RINGOVD                     | 0x19 | 0x039C9000:0x039E8FFF  | 0x20000      | d    | -     | same as size | None                                  |
+| WOFDATA                     | 0x1A | 0x039E9000:0x03CE8FFF  | 0x300000     | d    | -     | same as size | None                                  |
+| HB_VOLATILE                 | 0x1B | 0x03CE9000:0x03CEDFFF  | 0x5000       | d    | -     | same as size | None                                  |
+| [MEMD](#MEMD)               | 0x1C | 0x03cee000:0x03CFBFFF  | 0xE000       | d    | -     | same as size | VPD for RAM IIRC                      |
```
Comment (macpijan) [inline devnotes/pnor_layout.md:89]: @IgorBagnucki MEMD to be fixed here as well

File: devnotes/pnor_layout.md (hunk 8 above)
Comment (IgorBagnucki) [reply devnotes/pnor_layout.md]: I'm not sure, where this information can be found. For now I used description from pnor layout xml https://github.com/open-power/pnor/blob/master/p9Layouts/defaultPnorLayout_64.xml

File: devnotes/pnor_layout.md (hunk 11 above)
Comment (IgorBagnucki) [reply devnotes/pnor_layout.md]: Fixed

**Pull request merged**